torch/distributed/device_mesh.py

```diff
@@ -510,15 +510,16 @@ def __getitem__(
                 (mesh_dim_names,) if isinstance(mesh_dim_names, str) else mesh_dim_names
             )
 
+            it = iter(self.mesh_dim_names)
             if mesh_dim_names == self.mesh_dim_names:
                 return self
             elif len(mesh_dim_names) > len(self.mesh_dim_names) or not all(
-                mesh_dim_name in self.mesh_dim_names for mesh_dim_name in mesh_dim_names
+                mesh_dim_name in it for mesh_dim_name in mesh_dim_names
             ):
                 raise KeyError(
                     f"Invalid mesh_dim_name {mesh_dim_names} specified. "
-                    "Valid mesh_dim_names should be a subsequence of valid"
-                    f"mesh_dim_names from {self.mesh_dim_names}."
+                    "Valid mesh_dim_names should be a contiguous or non-contiguous "
```

Comment:
subsequence already contains both `contiguous` and `non-contiguous` cases. So the `contiguous or non-contiguous` is confusing and can be removed.